### PR TITLE
BLD: ipx/highs static libraries

### PR DESCRIPTION
**Pull request by mckib2**
#### Reference issue

https://github.com/rgommers/scipy/pull/107

#### What does this implement/fix?

Split ipx/highs into their own libraries

#### Additional information

The HiGHS upstream build system does this (splits everything into their own static libraries), we didn't do it originally in scipy because we found it exceedingly difficult to link against C++ distutil static libraries (C libraries are fine, C++ is where the issues were).  This is, however, simple to do in "proper" build systems like meson or cmake.

### CODE REVIEW

File: scipy/optimize/_highs/meson.build [hunk 1]
```diff
@@ -7,7 +7,12 @@ highs_define_macros = [
   '-DHIGHS_VERSION_MAJOR=1',  # don't care about this, look at CMakelists.txt
   '-DHIGHS_VERSION_MINOR=0',
   '-DHIGHS_VERSION_PATCH=0',
-  '-DHIGHS_DIR=' + meson.current_source_dir() / '..' / '..' / '_lib' / 'highs'
+  '-DHIGHS_DIR=' + meson.current_source_dir() / '..' / '..' / '_lib' / 'highs',
+  '-UOPENMP',
```
Comment (mckib2) [inline scipy/optimize/_highs/meson.build:11]: Figured it wouldn't hurt to have these everywhere and it's needed in multiple libraries now.  Please refactor into another list if desired

Comment (rgommers) [reply scipy/optimize/_highs/meson.build:11]: This seems fine to me.

File: scipy/optimize/_highs/meson.build [hunk 2]
```diff
@@ -51,171 +56,196 @@ basiclu_lib = static_library('basiclu',
...
 
 if cc.has_argument('-Wno-unused-but-set-variable')
   Wno_unused_but_set = ['-Wno-unused-but-set-variable']
 else
   Wno_unused_but_set = []
 endif
 
-_highs_wrapper = py3.extension_module('_highs_wrapper',
-  ['cython/src/_highs_wrapper.pyx', highs_sources, ipx_sources],
+highs_flags = ['-Wno-sign-compare', '-Wno-switch', '-Wno-format-truncation', '-Wno-non-virtual-dtor', '-Wno-class-memaccess']
```
Comment (mckib2) [inline scipy/optimize/_highs/meson.build:66]: Another case of putting in a list to apply to multiple libraries -- please refactor if desired

**Review by rgommers — APPROVED** [review]
Looks like a good idea to me, thanks. Let's give this a go!

**Pull request merged**